### DISCUSSION TITLE
Running rain template for JakartaEE on Glassfish 7.0.11

### DISCUSSION THREAD

**Discussion by kwee27** [primefaces/community#907] [PrimeFaces Templates] [answered]
Hi ... just looking for some pointers on deploying the rain template and demo onto glassfish 7.0.11 - using Jakarta EE10. The downloaded code has references to javax.faces and javax.servlet etc, which need to be changed to jakarta.faces and jakarta.servlet etc. 

Just looking for some basic docs to help get the process moving.

**Comment by kwee27** [comment]
So, the demo only runs on java 11 when using `mvn jetty:run`

**Comment by ancgate** [comment, marked as answer]
Check this out https://github.com/orgs/primefaces/discussions/356 and this one https://github.com/orgs/primefaces/discussions/742

**Reply by kwee27** [reply]
I worked through the various `javax.xxx` to `jakarta.xxx` refactorings (the project `pom` has the `shade` mappings which were helpful). Replaced the `@ManagedBean` annotation with a combination of `jakarta.inject.Named` and `jakarta.enterprise.context.ApplicationScoped`.

Also replaced the maven dependencies except the `poi` and `openpdf` ones with:
```
        <dependency>
            <groupId>jakarta.platform</groupId>
            <artifactId>jakarta.jakartaee-web-api</artifactId>
            <version>10.0.0</version>
            <scope>provided</scope>
        </dependency>
        <dependency>
            <groupId>org.primefaces</groupId>
            <artifactId>primefaces</artifactId>
            <version>13.0.4</version>
            <classifier>jakarta</classifier>
        </dependency>
```     

And removed this from the `web.xml`:
```
    <listener>
        <listener-class>org.apache.webbeans.servlet.WebBeansConfigurationListener</listener-class>
    </listener>
```

These changes break the embedded jetty server deployment, but that's fine for me. The demo is running on glassfish 7.0.11 with the `rain` template.

Thanks for the links @ancgate   

FYI ... this was also helpful:
[https://github.com/OmniFish-EE/upgrading-jakarta-ee-applications](url)
